group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -2080,6 +2715,55 @@ private <T> CoordinatorResult<T, CoordinatorRecord> consumerGroupFenceMember(
         }
     }
 
+    /**
+     * Removes a member from a share group.
+     *
+     * @param group       The group.
+     * @param member      The member.
+     *
+     * @return A list of records to be applied to the state.
+     */
+    private <T> CoordinatorResult<T, CoordinatorRecord> shareGroupFenceMember(
+        ShareGroup group,
+        ShareGroupMember member,
+        T response
+    ) {
+        List<CoordinatorRecord> records = new ArrayList<>();
+        List<CoordinatorRecord> replayRecords = new ArrayList<>();
+
+        replayRecords.add(newCurrentAssignmentTombstoneRecord(group.groupId(), member.memberId()));
+        replayRecords.add(newTargetAssignmentTombstoneRecord(group.groupId(), member.memberId()));
```

Comment:
This does not seem necessary as we could rely on the persisted record to delete the entire member. Could we?